assets/src/components/customers/CustomerDetailsSidebar.tsx

```diff
@@ -0,0 +1,316 @@
+                <CustomerDetailsProperty
+                  Icon={InfoCircleOutlined}
+                  key={key}
+                  name={key}
+                  value={value}
+                />
+              ))}
+            </CustomerDetailsSection>
+            <Divider dashed />
+          </>
+        )}
+
+        <CustomerDetailsSection title="Tags">
+          <SidebarCustomerTags customerId={customerId} />
+        </CustomerDetailsSection>
+      </Box>
+    </CustomerDetailsCard>
+  );
+};
+
+const CompanyDetailsSection = ({company}: {company: Company | undefined}) => {
+  let body;
+
+  if (company) {
+    const {
+      id: companyId,
+      name,
+      website_url: websiteUrl,
+      slack_channel_id: slackChannelId,
+      slack_channel_name: slackChannelName,
+    } = company;
+
+    body = (
+      <>
+        <CustomerDetailsProperty
+          Icon={TeamOutlined}
+          name="Name"
+          value={<Link to={`/companies/${companyId}`}>{name}</Link>}
+        />
+        <CustomerDetailsProperty
+          Icon={LinkOutlined}
+          name="Website"
+          value={websiteUrl}
+        />
+        <CustomerDetailsProperty
+          Icon={LinkOutlined}
+          name="Slack Channel"
+          value={
+            slackChannelId &&
+            slackChannelName && (
+              <a
+                href={`https://slack.com/app_redirect?channel=${slackChannelId}`}
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                {slackChannelName}
+              </a>
+            )
+          }
+        />
+      </>
+    );
+  } else {
+    // TODO: add ability to link customer to company
+    body = <Text>Customer is not linked to a company.</Text>;
+  }
+
+  return (
+    <CustomerDetailsSection title="Company">{body}</CustomerDetailsSection>
+  );
+};
+
+export const CustomerDetailsSection = ({
+  title,
+  children,
+}: {
+  title: string;
+  children: React.ReactNode;
+}) => {
+  return (
+    <Box>
+      <Box mb={2}>
+        <Text strong>{title}</Text>
+      </Box>
+      {children}
+    </Box>
+  );
+};
+
+const CustomerDetailsProperty = ({
+  Icon,
+  name,
+  value,
+}: {
+  Icon: React.ComponentType<any>;
+  name: string;
+  value: any;
+}) => {
+  let valueComponent;
+
+  if (React.isValidElement(value)) {
+    valueComponent = value;
+  } else {
+    valueComponent = <Text>{value || 'Unknown'}</Text>;
+  }
+
+  return (
+    <Flex mb={1} sx={{alignItems: 'center'}}>
+      <Icon style={{color: colors.primary}} />
+      <Box
+        ml={2}
+        mr={2}
+        sx={{
+          maxWidth: '100%',
+          overflow: 'hidden',
+          whiteSpace: 'nowrap',
+          textOverflow: 'ellipsis',
+        }}
+      >
+        <Text type="secondary">{name}</Text>
+      </Box>
+      {valueComponent}
+    </Flex>
+  );
+};
+
+const getLastSeenValue = ({
+  isOnline,
+  lastSeenAt,
+}: {
+  isOnline: boolean | undefined;
```

Comment:
i _think_ in typescript it's a bit more idiomatic to use `isOnline?: boolean` instead of `isOnline: boolean | undefined` (i'm pretty sure they're the same thing, but i might be wrong)

Reply:
Yeah, you're right. In this case, it isn't a big deal so I can change it to shorthand it, but there's one case where it actually does make a difference.

```
const firstFn = ({a, b}: {a: string, b: string | undefined}) => {
    console.log({a, b});
};

const secondFn = ({a, b}: {a: string, b?: string}) => {
    console.log({a, b});
};
```

In `firstFn`, `b` is required, but can be a `string` or `undefined`. In `secondFn`, `b` isn't required, but if it's there, it can be either `string` or `undefined`. 

How often does it happen where a property is required, but can be `undefined`? I'm not sure, but it's good to know the distinction. Again, for this case, it doesn't matter 🙂

Reply:
ahh good point, i'd say if a property can be undefined than it's nice to have it not be required :)